src/core/server/dev/dev_config.ts

```diff
@@ -6,26 +6,11 @@
  * Side Public License, v 1.
  */
 
-import { schema, TypeOf } from '@kbn/config-schema';
+import { schema } from '@kbn/config-schema';
 
 export const config = {
   path: 'dev',
-  schema: schema.object({
-    basePathProxyTarget: schema.number({
-      defaultValue: 5603,
-    }),
-  }),
+  // dev configuration is validated by the dev cli.
+  // we only need to register the `dev` schema to avoid failing core's config validation
```

Comment:
We are validating that no unknown key is present in the configuration file during core's config validation. Not adding this schema would cause this validation to fail when `dev.basePathProxyTarget` is present, for example.